repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/resource/AbstractResourcePreProcessor.java

```diff
@@ -76,7 +76,7 @@ void authorizeResourceUpdate(AtlasEntity resourceEntity, AtlasVertex ResourceVer
                 if (atlasVertexIterator.hasNext()) {
                     //Found linked asset in store
                     AtlasVertex assetVertex = (AtlasVertex) atlasVertexIterator.next();
-                    assetEntity = entityRetriever.toAtlasEntityHeader(assetVertex);
+                    assetEntity = new AtlasEntityHeader(entityRetriever.toAtlasEntity(assetVertex));
```

Comment:
Use function `toAtlasEntityHeaderWithClassifications` to fetch entities, why I'm saying, fetching whole entitiy sometimes slows it down if large no of relation present, which might be escalated later on. We should avoid using `toAtlasEntity` function and use alternative whenever possible.

Reply:
Nice catch, I agree